### PR TITLE
docs: update distributed.py

### PR DESCRIPTION
**Pull request by Tengxu-Sun**

### CODE REVIEW

File: bagua/torch_api/distributed.py [hunk 1]
```diff
@@ -60,25 +60,52 @@ def __init__(
             self.parameters_to_ignore = []
 
     def unwarp(self):
+        """
+        Return the unwarmped module.
```
Comment (NOBLES5E) [inline bagua/torch_api/distributed.py:64]: typo

Comment (NOBLES5E) [reply bagua/torch_api/distributed.py:64]: also the function name should be unwrap

Comment (Tengxu-Sun) [reply bagua/torch_api/distributed.py:64]: OK，this has been modified locally

File: bagua/torch_api/distributed.py [hunk 2]
```diff
@@ -60,25 +60,52 @@ def __init__(
...
+          (3) mark_bucket_ready
+          (4) mark_on_complete
+
+        2. bucket_initialized is True:
+          (1) mark_tensor_ready
+          (2) mark_on_complete 
+
+    Args:
+        module (torch.nn.Module): Module to be parallelized.
+        optimizers (list of optimizer(s)): Optimizer(s) can be single or mutiple. Its mainly used to divided the \
```
Comment (wangraying) [inline bagua/torch_api/distributed.py:96]: module is of type `DistributedModule` and optimizers are of type `torch.optim.Optimizer or list of torch.optim.Optimizers`

File: bagua/torch_api/distributed.py [hunk 3]
```diff
@@ -60,25 +60,52 @@ def __init__(
...
+                                        parameter into different bucets according to the optimizer(s) group.
+        bucket_type (BucketType): Type of elements in a communication bucket, could be \
+                    either module parameters, weights or gradients.
+        hierarchical_reduce (bool): Enable hierarchical reduce, which will perform an intra-node allreduce, followed \
+                                by an inter-node reduce defined by different `module`, and an intra-node broadcast at the end.
+        align_bytes (bool): Number to bytes to be aligned for each communication bucket.
+        chunking (bool): For scatter-gather communication pattern, set `chunking` to `True`.
+        fusion (bool): To reset parameter data pointer so that they can use faster code paths, set `fusion` to `True`.
+        decentralize_reduce (bool): Weather execute the decentralize communication.
```
Comment (wangraying) [inline bagua/torch_api/distributed.py:105]: typo: Weather -> Whether

File: bagua/torch_api/distributed.py (hunk 2 above)
Comment (Tengxu-Sun) [reply bagua/torch_api/distributed.py:96]: ok

File: bagua/torch_api/distributed.py [hunk 4]
```diff
@@ -522,6 +575,19 @@ def forward(self, *inputs, **kwargs):
 
 
 class ModelSwitchWrapper(torch.nn.Module):
+    """
+    Transform the original module into a distributed module.
```
Comment (shjwudp) [inline bagua/torch_api/distributed.py:579]: ModelSwitchWrapper should be regarded as a distributed algorithm wrapper, supporting different algorithm conversion, you can give a demo

File: bagua/torch_api/distributed.py (hunk 3 above)
Comment (Tengxu-Sun) [reply bagua/torch_api/distributed.py:105]: yes，'False'

File: bagua/torch_api/distributed.py (hunk 4 above)
Comment (Tengxu-Sun) [reply bagua/torch_api/distributed.py:579]: ok

File: bagua/torch_api/distributed.py [hunk 5]
```diff
@@ -737,6 +821,9 @@ def update_hyperparameters(rec_hp):
         return update_hyperparameters(recommended_hyperparameters)
 
     def forward(self, *inputs, **kwargs):
+        """
+        Overwritte the forward processs and return the output.
```
Comment (shjwudp) [inline bagua/torch_api/distributed.py:825]: typo: Overwrite

File: bagua/torch_api/distributed.py (hunk 1 above)
Comment (NOBLES5E) [reply bagua/torch_api/distributed.py:64]: let me know when all conversations are resolved so that I can start another round of review

Comment (Tengxu-Sun) [reply bagua/torch_api/distributed.py:64]: ok

File: bagua/torch_api/distributed.py [hunk 6]
```diff
@@ -59,32 +59,63 @@ def __init__(
...
-    bucket_initialized is False:
-        1. add_param
-        2. initialize_buckets -> register_models
-        3. mark_bucket_ready
-        4. mark_on_complete
-
-    bucket_initialized is True:
-        1. mark_tensor_ready
-        2. mark_on_complete
+    r"""In order to improve communication efficiency, the distributed algorithm divides the parameter into different blocks,
```
Comment (NOBLES5E) [inline bagua/torch_api/distributed.py:79]: algorithm "chunks parameters into many buckets. A bucket is the minimum unit of communication between devices in bagua."

File: bagua/torch_api/distributed.py [hunk 7]
```diff
@@ -59,32 +59,63 @@ def __init__(
...
-        2. initialize_buckets -> register_models
-        3. mark_bucket_ready
-        4. mark_on_complete
-
-    bucket_initialized is True:
-        1. mark_tensor_ready
-        2. mark_on_complete
+    r"""In order to improve communication efficiency, the distributed algorithm divides the parameter into different blocks, 
+    called buckets, and the parameter gradient communicates between different devices in the form of buckets. 
+    This module is the bucket manager，providing bucket operation methods.
```
Comment (NOBLES5E) [inline bagua/torch_api/distributed.py:81]: need space after comma

File: bagua/torch_api/distributed.py [hunk 8]
```diff
@@ -59,32 +59,63 @@ def __init__(
...
+            1.4 mark_on_complete
+
+        2. bucket_initialized is True:
+            2.1 mark_tensor_ready
+
+            2.2 mark_on_complete 
+
+    Args:
+        module (DistributedModule): Module to be parallelized.
+        optimizers (torch.optim.Optimizer or list of torch.optim.Optimizer): Optimizer(s) can be single or mutiple. Its mainly used to
```
Comment (NOBLES5E) [inline bagua/torch_api/distributed.py:101]: optimizers contain one or more PyTorch optimizers.

Comment (NOBLES5E) [reply bagua/torch_api/distributed.py:101]: Remove "Its mainly ... group."

File: bagua/torch_api/distributed.py [hunk 9]
```diff
@@ -198,8 +232,13 @@ def fill_slot(self, param):
             return param
 
     def initialize_buckets(self) -> List[List[torch.Tensor]]:
-        """
-        NOTE: initialize_buckets MUST execute after the first round of backward for grad ready.
+        r"""
+        Initialize parameter buckets.
+
+        .. note:: Initialize_buckets MUST execute after the first round of backward for grad ready.
```
Comment (NOBLES5E) [inline bagua/torch_api/distributed.py:238]: remove `for grad ready`

File: bagua/torch_api/distributed.py (hunk 4 above)
Comment (shjwudp) [reply bagua/torch_api/distributed.py:584]: The function description is incomplete. ModelSwitchWrapper is designed to switch distributed algorithms during training process

Like this:

model = bagua_init(DistributedAlgorithm. GradientAllReduce)
... train A epoch
model.switch_to(DistributedAlgorithm.Decentralize)
... train B epoch
model.switch_to(DistributedAlgorithm. GradientAllReduce) 
... continue training

Comment (Tengxu-Sun) [reply bagua/torch_api/distributed.py:584]: ok, i‘ll supplement it more complete

**Review by shjwudp — DISMISSED** [review]

**Pull request merged**